leaflet-treering.js

```diff
@@ -348,19 +348,77 @@ function MeasurementData (dataObject) {
    * @function insertPoint
    */
   MeasurementData.prototype.insertPoint = function(latLng, hasLatewood) {
-    var i = 0;
-    while (this.points[i] != undefined &&
-        this.points[i].latLng.lng < latLng.lng) {
-      i++;
-    }
+    var disList = [];
+
+    /**
+    * calculate the distance between 2 points
+    * @function distanceCalc
+    * @param {first point.latLng} pointA
+    * @param {second point.latLng} pointB
+    */
+    function distanceCalc (pointA, pointB) {
+      return Math.sqrt(Math.pow((pointB.lng - pointA.lng), 2) +
+                       Math.pow((pointB.lat - pointA.lat), 2));
+    };
+
+    // finds point with smallest abs. distance
+    for (i = 0; i < this.points.length; i++) {
+      var distance = Number.MAX_SAFE_INTEGER;
+      if (this.points[i] && this.points[i].latLng) {
+         var currentPoint = this.points[i].latLng;
+         distance = distanceCalc(currentPoint, latLng);
+      disList.push(distance);
+    };
+
+    var minDistance = Math.min(...disList);
+    i = disList.indexOf(minDistance)
+
     if (this.points[i] == null) {
-      alert('New point must be within existing points.' +
-          'Use the create toolbar to add new points to the series.');
+      alert('New point must be within existing points. Use the create toolbar to add new points to the series.');
       return;
     }
 
+    // define 3 points: i, i - 1, i + 1
+    var iPoint = this.points[i].latLng;
+
+    if (this.points[i - 1].start) {
+      var iMinus = this.points[i].latLng;
+    } else {
+      var iMinus = this.points[i - 1].latLng;
+    };
+
+    if (this.points[i + 1]) { //
+      var iPlus = this.points[i + 1].latLng;
+    } else {
+      var iPlus = L.latLng(0.15, 0.6);
+    };
+
+    // distance: point[i] to point[i + 1]
+    var disIPlus = distanceCalc(iPoint, iPlus);
```

Comment:
all these variables are getting kinda confusingly named, including the latLng coming in to the method call. Maybe rename things to make more sense? newPoint and whatnot?